### PR TITLE
Make hook signature more open

### PR DESCRIPTION
**Pull request by yohanboniface**
Third-party extensions/users may use hooks for their own needs.

An example is utilery: https://github.com/tilery/utilery/blob/async/utilery/core.py#L70

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -60,7 +60,8 @@ A how-to guide:
 * [How to return JSON content](how-to-guides.md#how-to-return-json-content)
 * [How to subclass Roll itself](how-to-guides.md#how-to-subclass-roll-itself)
 * [How to deploy Roll into production](how-to-guides.md#how-to-deploy-roll-into-production)
-* [How to run Roll’s tests](how-to-guides.md##how-to-run-rolls-tests)
+* [How to run Roll’s tests](how-to-guides.md#how-to-run-rolls-tests)
+* [How to send custom events](how-to-guides.md#how-to-use-custom-events)
```
Comment (davidbgk) [inline docs/index.md]: `send` (here) or `use` as a title of the How-to page?

File: docs/how-to-guides.md [hunk 2]
```diff
@@ -211,3 +211,20 @@ your virtualenv active), run:
 Then you can run the tests:
 
     py.test
+
+
+## How to use custom events
+
+Roll as a very small API for listening and sending event. It's possible to use
```
Comment (davidbgk) [inline docs/how-to-guides.md]: has*

File: docs/how-to-guides.md [hunk 3]
```diff
@@ -211,3 +211,20 @@ your virtualenv active), run:
...
+it in your project for your own events.
+
+That would look like this:
+
+    app = Roll()
+
+    @app.listen('myevent')
+    def listener(myarg):
```
Comment (davidbgk) [inline docs/how-to-guides.md]: What would be a real use-case? Sending emails? Filling metrics?

File: docs/index.md (hunk 1 above)
Comment (yohanboniface) [reply docs/index.md]: Done.

File: docs/how-to-guides.md (hunk 2 above)
Comment (yohanboniface) [reply docs/how-to-guides.md]: Done.

File: docs/how-to-guides.md (hunk 3 above)
Comment (yohanboniface) [reply docs/how-to-guides.md]: Added a more realistic use case. :)

File: docs/how-to-guides.md [hunk 4]
```diff
@@ -211,3 +211,33 @@ your virtualenv active), run:
...
 
     py.test
+
+
+## How to send custom events
+
+Roll has a very small API for listening and sending events. It's possible to use
+it in your project for your own events.
+
+Events are useful when you want other user to extend your own code, whether
```
Comment (davidbgk) [inline docs/how-to-guides.md]: s/user/users/

File: docs/how-to-guides.md [hunk 5]
```diff
@@ -211,3 +211,33 @@ your virtualenv active), run:
...
+it's a Roll extension, or a full project built with Roll.
+They differ from configuration in that they are more adapted for dynamic
+modularity.
+
+For example, say we develop a DB pooling extension for Roll. We
+would use a simple configuration parameter to let users change the connection
+credentials (host, username, password…). But if we want users run some
```
Comment (davidbgk) [inline docs/how-to-guides.md]: s/run/to run/

Comment (yohanboniface) [reply docs/how-to-guides.md]: Fixed!

**Pull request merged**